src/topology/algebra/infinite_sum.lean

```diff
@@ -676,8 +676,14 @@ end
 section tsum
 variables [t2_space α]
 
-lemma tsum_neg (hf : summable f) : ∑'b, - f b = - ∑'b, f b :=
-hf.has_sum.neg.tsum_eq
+lemma tsum_neg : ∑'b, - f b = - ∑'b, f b :=
+begin
+  by_cases hf : summable f,
+  { exact hf.has_sum.neg.tsum_eq, },
+  { rw [tsum_eq_zero_of_not_summable hf, neg_zero],
+    rw ← summable_neg_iff at hf,
+    rw tsum_eq_zero_of_not_summable hf },
+end
```

Comment:
I wrote something very much like this for `matrix.transpose_tsum`:

https://github.com/leanprover-community/mathlib/blob/32320a100c580e1f5b0339ebf03c2b8d8a92fe90/src/topology/instances/matrix.lean#L261-L266

So I guess this can be golfed to
```suggestion
begin
  by_cases hf : summable f,
  { exact hf.has_sum.neg.tsum_eq, },
  { have hfn := summable_neg_iff.mpr hf,
    rw [tsum_eq_zero_of_not_summable hf, tsum_eq_zero_of_not_summable hfn, neg_zero] },
end
```
(untested)

Reply:
If you want to avoid the `have`, you can use
```
  { rw [tsum_eq_zero_of_not_summable hf, tsum_eq_zero_of_not_summable (mt summable.of_neg hf),
      neg_zero] },
```
or even just
```
{ simp [tsum_eq_zero_of_not_summable hf, tsum_eq_zero_of_not_summable (mt summable.of_neg hf)] },
```

Reply:
Yeah I like the `simp` solution best. Thanks! Nice `mt`.